src/core/cosmopolite.scala

```diff
@@ -33,12 +33,15 @@ object Messages:
       Messages[L](Map(summon[ValueOf[L]].value -> string))
 
 case class Messages[L <: String](text: Map[String, String]):
-   def &[L2 <: String](messages: Messages[L2])(using NotGiven[(L | L2) =:= L]): Messages[L | L2] =
+   def &[L2 <: String](messages: Messages[L2])(using NotGiven[L2 <:< L]): Messages[L | L2] =
```

Comment:
I'm not sure how this test is different from the one I had. It seems like both would be problematic if trying to combine, say `En | Fr` with `Fr | De`.

Reply:
They should be equivalent, but the original constraint did fail preventing it. There must be a bug in the interaction of Notgiven implicit search and the type expression reduction.

Reply:
Regarding combining, that also works. This works:
```scala
   val msg: Messages[MyLangs] =
      en"This is ${number(1)} in English" &
      de"Das ist ${number(1)} auf Deutsch" &
      es"Es ${number(1)} en español" &
      fr"C'est ${number(1)} en français"

   val msg2: Messages[Pt | It] =
      pt"This is ${number2(1)} in port" &
      it"Das ist ${number2(1)} it"

   val combined: Messages[MyLangs | Pt | It] = msg & msg2
```

Reply:
Yes, regarding the combination with overlap, both casses will miss it